zaza/openstack/charm_tests/neutron/tests.py

```diff
@@ -29,6 +30,206 @@
 import zaza.openstack.utilities.openstack as openstack_utils
 
 
+class NeutronGatewayTest(test_utils.OpenStackBaseTest):
+    """Test basic Neutron Gateway Charm functionality."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Run class setup for running Neutron Gateway tests."""
+        super(NeutronGatewayTest, cls).setUpClass()
+        cls.current_os_release = openstack_utils.get_os_release()
+        cls.services = cls._get_services()
+
+        # set up clients
+        cls.neutron_client = (
+            openstack_utils.get_neutron_session_client(cls.keystone_session))
+
+        bionic_stein = openstack_utils.get_os_release('bionic_stein')
+
+        cls.pgrep_full = (True if cls.current_os_release >= bionic_stein
+                          else False)
+
+    def test_400_create_network(self):
+        """Create a network, verify that it exists, and then delete it."""
+        logging.debug('Creating neutron network...')
+        self.neutron_client.format = 'json'
+        net_name = 'test_net'
+
+        # Verify that the network doesn't exist
+        networks = self.neutron_client.list_networks(name=net_name)
+        net_count = len(networks['networks'])
+        assert net_count == 0, (
+            "Expected zero networks, found {}".format(net_count))
+
+        # Create a network and verify that it exists
+        network = {'name': net_name}
+        self.neutron_client.create_network({'network': network})
+
+        networks = self.neutron_client.list_networks(name=net_name)
+        logging.debug('Networks: {}'.format(networks))
+        net_len = len(networks['networks'])
+        assert net_len == 1, (
+            "Expected 1 network, found {}".format(net_len))
+
+        logging.debug('Confirming new neutron network...')
+        network = networks['networks'][0]
+        assert network['name'] == net_name, "network ext_net not found"
+
+        # Cleanup
+        logging.debug('Deleting neutron network...')
+        self.neutron_client.delete_network(network['id'])
+
+    def test_401_enable_qos(self):
+        """Check qos settings set via neutron-api charm."""
+        if (self.current_os_release >=
+                openstack_utils.get_os_release('trusty_mitaka')):
+            logging.info('running qos check')
+
+            with self.config_change(
+                    {'enable-qos': 'False'},
+                    {'enable-qos': 'True'},
+                    application_name="neutron-api"):
+
+                # wait for neutron-gateway to complete as well
+                time.sleep(60)
```

Comment:
The time.sleep may have been in amulet. But we have a standard for zaza that we never use these. See tenacity in various other tests as a way of waiting or retrying.

Reply:
done - [commit](https://github.com/openstack-charmers/zaza-openstack-tests/pull/142/commits/f79f2fb54c062651324f7140e9f90941c565c302)

Validated via func-smoke. Tenacity is nice for sure :)